src/CommandingOfficers/CommanderAve.java

```diff
@@ -0,0 +1,534 @@
+      myCommander.addCOModifier(damageMod);
+      myCommander.addCOModifier(snowSpreadMod);
+
+      // Keep track of any tiles that change to snow.
+      GameEventQueue glacioEvents = new GameEventQueue();
+
+      // Add extra snow to any tile that already has snow, plus extra on owned buildings.
+      for( int x = 0; x < Ave.snowMap.length; ++x )
+        for( int y = 0; y < Ave.snowMap[0].length; ++y )
+        {
+          Ave.snowMap[x][y] *= GLACIO_SNOW_MULTIPLIER;
+        }
+      Ave.addSnow(SNOW_PER_TURN*GLACIO_SNOW_MULTIPLIER, gameMap, glacioEvents);
+      Ave.relevelSnow(gameMap, glacioEvents);
+
+      // Change terrain to snow around each of Ave's units, and damage trees and enemies.
+      HashSet<XYCoord> tilesSeen = new HashSet<XYCoord>();
+      for( Unit unit : myCommander.units )
+      {
+        // Two-tile radius of effect around each unit.
+        ArrayList<XYCoord> tiles = Utils.findLocationsInRange(gameMap, new XYCoord(unit.x, unit.y), 0, 2);
+        for( XYCoord coord : tiles )
+        {
+          // Don't Glacio the same tile twice.
+          if( tilesSeen.contains(coord) ) continue;
+          tilesSeen.add(coord);
+
+          // Add snow around friendlies.
+          Location loc = gameMap.getLocation(coord);
+
+          // Turn the spaces around each unit to snow.
+          Environment tileEnvi = loc.getEnvironment();
+          if((tileEnvi.weatherType != Weathers.SNOW) || (tileEnvi.terrainType == TerrainType.FOREST))
+          {
+            // Destroy any forests. Big hail, man.
+            TerrainType newTerrain = (loc.getEnvironment().terrainType == TerrainType.FOREST) ? TerrainType.GRASS : loc.getEnvironment().terrainType;
+            Environment newEnvi = Environment.getTile(newTerrain, Weathers.SNOW);
+            GameEvent event = new MapChangeEvent(coord, newEnvi);
+            glacioEvents.add(event);
+          }
+
+          // Damage each enemy nearby.
+          if( null != loc.getResident() && myCommander.isEnemy(loc.getResident().CO) )
+          {
+            loc.getResident().alterHP(-2);
+          }
+        }
+      }
+
+      // Do all of our terrain alterations.
+      while(!glacioEvents.isEmpty())
+      {
+        GameEvent event = glacioEvents.poll();
+        event.performEvent(gameMap);
+        GameEventListener.publishEvent(event);
+      }
+    }
+  } // ~Glacio
+
+  public static CommanderInfo getInfo()
+  {
+    return coInfo;
+  }
+
+  private static class SnowPail implements Comparable<SnowPail>
+  {
+    public final XYCoord snowCoord;
+    public final int snowDepth;
+
+    public SnowPail(XYCoord coord, int snowAmt)
+    {
+      snowCoord = coord;
+      snowDepth = snowAmt; // Truncate to two-digit precision.
+    }
+
+    @Override
+    public int compareTo(SnowPail other)
+    {
+      // Sort them shallowest first.
+      return snowDepth - other.snowDepth;
+    }
+  }
+
+  private static class SnowSpreadDistanceModifier implements COModifier
```

Comment:
I don't know how much I like the idea of making this a COMod rather than just having the ability boost it and the CO reset it to default on turn init.
It feels like a large gain in LoC without a commensurate boost in correctness/functionality.
Thoughts?

Reply:
Fair. I might do that. It's a turn-on turn-off thing, which `COModifiers` are made to do, so I didn't put too much thought into it.